### PR TITLE
Added tests for cylindrical coordinates.

### PR DESCRIPTION
**Pull request by mfarrington1**
## Related issues

Added tests for cylindrical coordinates and optimized some functions in PtEtaPhiM.jl

## Checklist

- [x] Tests are passing
- [x] Docs were updated and workflow is passing
- [ ] Add edge case for eta

### CODE REVIEW

**Comment by szabo137** [comment]
Could you please rebase and push again?

**Comment by szabo137** [comment]
@mfarrington1 I think its better to just rebase to the current main, instead of adding the PRs by hand, because the latter messes up the history. 

If you like, I can do that for you?

**Comment by szabo137** [comment]
@mfarrington1 I successfully rebased and formatted locally. Is it ok if I just force push, so we can proceed with this one?

**Comment by mfarrington1** [comment]
> @mfarrington1 I successfully rebased and formatted locally. Is it ok if I just force push, so we can proceed with this one?

Yup, thanks for handling this! Let me know if there are any issues.

File: src/coordinate_systems/PtEtaPhiM.jl [hunk 1]
```diff
@@ -92,15 +92,9 @@ end
...
-  xcomp = px(mom)
-  ycomp = py(mom)
-  zcomp = pz(mom)
-
-  return if iszero(xcomp) && iszero(ycomp) && iszero(zcomp)
-    zero(xcomp)
-  else
-    atan(transverse_momentum(mom), zcomp)
-  end
+  η = eta(mom)
```
Comment (szabo137) [inline src/coordinate_systems/PtEtaPhiM.jl:95]: What is the reason for the change in the implementation? The new implementation performs a square root, which might cause inaccuracies.

File: test/runtests.jl [hunk 2]
```diff
@@ -10,4 +10,8 @@ begin
   @time @safetestset "xyze" begin
     include("xyze.jl")
   end
+
+  @time @safetestset "ptetaphim" begin
```
Comment (szabo137) [inline test/runtests.jl:14]:
```suggestion
  @time @safetestset "PtEtaPhiM" begin
```

**Review by szabo137 — APPROVED** [review]
LGFM. Just two little comments. Feel free to merge after reply.

File: src/coordinate_systems/PtEtaPhiM.jl (hunk 1 above)
Comment (mfarrington1) [reply src/coordinate_systems/PtEtaPhiM.jl:95]: Sorry, maybe I'm misunderstanding, how is the new implementation performing a square root? The change for this implementation was because it was more succinct and did not require switching into Cartesian coordinates to find the polar angle, which feels a bit clunky. However, if this change makes things less accurate then there is no reason to keep things as they were.

Comment (szabo137) [reply src/coordinate_systems/PtEtaPhiM.jl:95]: I understand your point about the cartesian coordinates, especially the `sincos` of `phi`.

The `spatial_magnitude` function performs the `sqrt`, which might lead to inaccuracies. However, I just realized, that it is just used in an `iszero` check, which means, just `iszero(pt(mom))` (which is the only way to make `r==0`) would also do the trick, right?

Comment (mfarrington1) [reply src/coordinate_systems/PtEtaPhiM.jl:95]: Ah, I see what you are saying. This was done to copy the ROOT implementation of `polar_angle()`,  

` T Theta() const { using std::atan; return fRho > 0 ? 2 * atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }`

which you can find here https://root.cern/doc/master/GenVector_2CylindricalEta3D_8h_source.html#l00160

Having said this, I also see that what I wrote is slightly wrong insofar as it doesn't capture what to do when eta is greater or less than 0 like ROOT does. So maybe I need to change this if you think this definition is the one we want to use.

Comment (szabo137) [reply src/coordinate_systems/PtEtaPhiM.jl:95]: Agreed 👍
